src/lib/components/Card/Card.tsx

```diff
@@ -22,29 +23,40 @@ export interface FlowbiteCardImageTheme {
   horizontal: FlowbiteBoolean;
 }
 
-export interface CardProps extends PropsWithChildren<ComponentProps<'div'>> {
+type CommonCardProps = PropsWithChildren<ComponentProps<'div'>> & {
   horizontal?: boolean;
   href?: string;
-  imgAlt?: string;
-  imgSrc?: string;
   theme?: DeepPartial<FlowbiteCardTheme>;
-}
+};
+
+export type CardProps =
+  | (
+      | { imgAlt?: string; imgSrc?: string; renderImage?: never }
+      | {
+          renderImage?: () => JSX.Element;
+          imgAlt?: never;
+          imgSrc?: never;
+        }
+    ) &
+      CommonCardProps;
 
-export const Card: FC<CardProps> = ({
-  children,
-  className,
-  horizontal,
-  href,
-  imgAlt,
-  imgSrc,
-  theme: customTheme = {},
-  ...props
-}) => {
+export const Card: FC<CardProps> = ({ children, className, horizontal, href, theme: customTheme = {}, ...props }) => {
   const Component = typeof href === 'undefined' ? 'div' : 'a';
-  const theirProps = props as object;
+  const theirProps = omit(props, ['renderImage', 'imgSrc', 'imgAlt']) as object;
 
   const theme = mergeDeep(useTheme().theme.card, customTheme);
 
+  const Image = () =>
+    'renderImage' in props && props.renderImage ? (
+      props.renderImage()
+    ) : 'imgSrc' in props ? (
+      <img
+        alt={props.imgAlt ?? ''}
+        src={props.imgSrc}
+        className={classNames(theme.img.base, theme.img.horizontal[horizontal ? 'on' : 'off'])}
+      />
+    ) : null;
```

Comment:
I am not such a big fan of such ternary magic. One could pull this out into a couple of new components but then also that might get a little out of hand. Please advise.

Reply:
You could achieve this via null coalescing instead? Also you've changed the behaviour by doing a check if imgSrc is present, when before it was only checking if it was truthy, which I have rolled back.

```suggestion
  const Image = () => props.renderImage?.() ??  props.imgSrc ? (
      <img
        alt={props.imgAlt ?? ''}
        src={props.imgSrc}
        className={classNames(theme.img.base, theme.img.horizontal[horizontal ? 'on' : 'off'])}
      />
    ) : null;
```

Reply:
Did you check that this works with typescript? About the `imgSrc`: Are you saying that now it would render an image, if someone set `imgSrc` to `false` and that is a problem? Maybe one should add a test for that... I will give it some love and consider your remarks.

Overall I have to say this is very, very hard to read and make sense of and the coalescing is "cool" but adding mental load imo.

Reply:
This is outdate, so, I'll mark it as resolved.